src/shims/unix/foreign_items.rs

```diff
@@ -315,6 +315,41 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
                     }
                 }
             }
+            "aligned_alloc" => {
+                let [align, size] =
+                    this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+                let align = this.read_target_usize(align)?;
+                let size = this.read_target_usize(size)?;
+
+                // The size must be a power of the alignment.
+                // The specs says nothing about error handling other than null
+                // pointer but aligned_alloc Linux/macOs implementations set EINVAL. Note that
+                // on FreeBSD setting a size not being a power of alignment is UB but we do not
+                // want it.
+                //
+                // http://en.cppreference.com/w/cpp/memory/c/aligned_alloc
+                // Linux: https://linux.die.net/man/3/aligned_alloc
+                // FreeBSD: https://man.freebsd.org/cgi/man.cgi?query=aligned_alloc&apropos=0&sektion=3&manpath=FreeBSD+9-current&format=html
+                match size.checked_rem(align) {
+                    Some(0) => {
```

Comment:
```suggestion
                    Some(0) if align.is_power_of_two() && align >= this.pointer_size().bytes() => {
```
Then you need only one error path

Reply:
Probably the better way to do this is a little helper function `is_align_supported` or so.